crates/ursa-tracker/src/ip_api.rs

```diff
@@ -0,0 +1,99 @@
+use anyhow::{anyhow, Result};
+use geohash::encode;
+use hyper::{
+    client::connect::dns::GaiResolver,
+    service::Service,
+    Client
+};
+use hyper_tls::HttpsConnector;
+use serde_derive::Deserialize;
+
+#[derive(Deserialize, Default, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase", default)]
+pub struct IpInfoResponse {
+    #[serde(rename = "ip")]
+    pub addr: String,
+    pub hostname: String,
+    pub city: String,
+    pub region: String,
+    pub country: String,
+    pub loc: String,
+    pub org: String,
+    pub postal: String,
+    pub timezone: String,
+    pub geo: String,
+}
+
+/// Get public ip info from https://ipinfo.io
+pub async fn get_ip_info(token: String, addr: String) -> Result<IpInfoResponse> {
```

Comment:
I think it would be nicer if the `ipinfo.io`'s type was not publicly exposed as the service which is used should be an implementation detail, and we had an owned struct which we could rely on its format without the underlying service.

Reply:
Since pretty much all ip apis have very similar responses, its pretty easy to adjust the field names and continue.